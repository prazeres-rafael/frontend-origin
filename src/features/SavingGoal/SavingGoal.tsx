import React from 'react';
import Card from '../../components/Card';
import BuyHouse from '../../assets/svg/buy-a-house.svg';
import {
  AmountDescription,
  AmountParagraph,
  AmountSection,
  AmountValue,
  ConfirmSection,
  GoalSection,
  Paragraph,
  PaperBase,
  ResultSection,
  Title,
  SavingGoalParagraph,
  Section,
  Subtitle,
} from './styled';
import { Box, Grid } from '@mui/material';
import Input from '../../components/Input';
import MonthPicker from '../../components/MonthPicker';
import Cta from '../../components/Cta';

export default function Header(): React.ReactElement {
  return (
    <Section>
      <SavingGoalParagraph>
        Let‘s plan your <b>saving goal.</b>
      </SavingGoalParagraph>
      <Card>
        <GoalSection>
          <BuyHouse />
          <Box>
            <Title>Buy a house</Title>
            <Subtitle>Saving goal</Subtitle>
          </Box>
        </GoalSection>
        <AmountSection>
          <Grid container spacing={{ lg: 2, md: 2 }}>
            <Grid item lg={7} md={7} xs={12}>
              <Paragraph>Total amount</Paragraph>
              <Input />
            </Grid>
            <Grid item lg={5} md={5} xs={12}>
              <Paragraph>Reach goal by</Paragraph>
              <MonthPicker />
            </Grid>
          </Grid>
        </AmountSection>
        <ResultSection>
          <PaperBase elevation={0}>
            <Box>
              <AmountParagraph>Monthly amount</AmountParagraph>
              <AmountValue>$520.83</AmountValue>
            </Box>
            <AmountDescription>
              You‘re planning <b>48 monthly deposits</b> to reach your
              <b> $25,000</b> goal by <b>October 2020</b>.
            </AmountDescription>
          </PaperBase>
          <ConfirmSection>
            <Cta>Confirm</Cta>
          </ConfirmSection>
        </ResultSection>
      </Card>
    </Section>
  );
}
