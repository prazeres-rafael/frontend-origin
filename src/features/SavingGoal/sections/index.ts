export { default as Amonth } from './Amonth';
