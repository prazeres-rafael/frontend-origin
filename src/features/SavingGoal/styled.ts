import styled from 'styled-components';
import { Typography, Paper } from '@mui/material';

export const SavingGoalParagraph = styled(Typography)({
  color: '#1B31A8',
  fontSize: 20,
  fontWeight: 500,
  paddingTop: 48,
  marginBottom: 24,
});

export const Title = styled(Typography)({
  color: '#1E2A32',
  fontSize: 24,
  fontWeight: 500,
});

export const Subtitle = styled(Typography)({
  color: '#708797',
  fontSize: 16,
  fontWeight: 400,
});

export const AmountParagraph = styled(Typography)({
  color: '#1E2A32',
  fontSize: 20,
  fontWeight: 400,
  marginLeft: 32,
});

export const AmountValue = styled(Typography)({
  color: '#0079FF',
  margin: '24px 30.5px 16px',
  fontSize: 32,
  fontWeight: 500,
  textAlign: 'right',
});

export const AmountDescription = styled(Typography)({
  color: '#1C1E1F',
  margin: '24px 32px',
  fontSize: 12,
  fontWeight: 400,
});

export const Paragraph = styled(Typography)({
  color: '#1E2A32',
  fontSize: 14,
  fontWeight: 400,
});

export const Section = styled.section({
  backgroundColor: '#F4F8FA',
  height: '90vh',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
});

export const GoalSection = styled.div({
  display: 'flex',
  flexDirection: 'row',

  '& > div': {
    marginLeft: 16,
  },
});

export const AmountSection = styled.div({
  display: 'flex',
  flexDirection: 'row',
  marginTop: '28px',

  '& > div:last-child': {
    marginLeft: 16,
  },
});

export const ResultSection = styled.div({
  marginTop: '24px',
});

export const PaperBase = styled(Paper)({
  border: '1px solid #E9EEF2',
  borderRadius: 8,
  background: 'linear-gradient(180deg, #FFF 50%, #F4F8FA 50%)',

  '& > div': {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});

export const ConfirmSection = styled.div({
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'center',
  marginTop: '32px',
});
