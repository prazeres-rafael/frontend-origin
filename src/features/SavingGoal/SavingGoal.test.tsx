import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import React from 'react';

import SavingGoal from './index';

const months = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

describe('SavingGoal', () => {
  it('should match the snapshot', () => {
    const { container } = render(<SavingGoal />);

    expect(container).toMatchSnapshot();
  });

  it('should not decrement if date is less than current date', () => {
    render(<SavingGoal />);

    userEvent.click(screen.getByRole('buttonDecrement'));
    const currentMonth = new Date().getMonth();
    const currentYear = new Date().getFullYear();

    expect(screen.getByRole('paragraphMonth')).toHaveTextContent(
      months[currentMonth]
    );
    expect(screen.getByRole('paragraphYear')).toHaveTextContent(
      String(currentYear)
    );
  });

  it('should increment month and year after buttonIncrement click', () => {
    render(<SavingGoal />);

    const incrementButton = screen.getByRole('buttonIncrement');
    userEvent.click(incrementButton);
    const date = new Date();
    date.setMonth(date.getMonth() + 1);

    expect(screen.getByRole('paragraphMonth')).toHaveTextContent(
      months[date.getMonth()]
    );
    expect(screen.getByRole('paragraphYear')).toHaveTextContent(
      String(date.getFullYear())
    );
  });

  it('should increment month and year using keyboard when the component is focused', () => {
    render(<SavingGoal />);

    const monthContent = screen.getByRole('monthContent');
    monthContent.focus();
    userEvent.keyboard('{arrowright}');
    const date = new Date();
    date.setMonth(date.getMonth() + 1);

    expect(screen.getByRole('paragraphMonth')).toHaveTextContent(
      months[date.getMonth()]
    );

    userEvent.keyboard('{arrowleft}');
    date.setMonth(new Date().getMonth());

    expect(screen.getByRole('paragraphMonth')).toHaveTextContent(
      months[date.getMonth()]
    );
  });

  it('should change and calculate the monthly amount onChange of input', () => {
    render(<SavingGoal />);

    userEvent.type(screen.getByRole('textbox'), '100');

    expect(screen.getByRole('amountValue')).toHaveTextContent('100');
  });

  it('should change and calculate the monthly amount onChange of monthPicker', () => {
    render(<SavingGoal />);

    userEvent.type(screen.getByRole('textbox'), '100');
    const incrementButton = screen.getByRole('buttonIncrement');
    userEvent.click(incrementButton);

    expect(screen.getByRole('amountValue')).toHaveTextContent('50');
  });

  it('should change the amount description onChange of input and monthPicker', () => {
    render(<SavingGoal />);

    userEvent.type(screen.getByRole('textbox'), '100');

    expect(screen.getByRole('amountDescription')).toHaveTextContent('100');

    const incrementButton = screen.getByRole('buttonIncrement');
    userEvent.dblClick(incrementButton);

    expect(screen.getByRole('amountDescription')).toHaveTextContent(
      '3 monthly'
    );
  });
});
