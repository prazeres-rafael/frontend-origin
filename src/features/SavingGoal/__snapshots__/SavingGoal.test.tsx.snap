// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavingGoal should match the snapshot 1`] = `
.c2 {
  box-shadow: 0px 16px 32px rgba(30,42,50,0.08);
  border-radius: 8px;
  width: 100%;
}

.c3 {
  padding: 40px 24px;
}

.c3:last-child {
  padding-bottom: 40px;
}

.c19 {
  margin: 18px 128px;
  font-family: Work Sans,sans-serif;
  text-transform: none;
  font-size: 16px;
  font-weight: 600;
}

.c18 {
  background-color: #1B31A8;
  border-radius: 32px;
}

.c10 {
  color: #1E2A32;
  font-size: 16px;
  font-weight: 600;
  text-align: center;
  margin: 3px 0 2px 0;
}

.c11 {
  color: #708797;
  font-size: 16px;
  font-weight: 400;
  text-align: center;
  margin: 2px 0 3px 0;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E9EEF2;
  border-radius: 4px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c9 > div {
  margin: 4px 0;
}

.c9 > button {
  cursor: pointer;
  background-color: transparent;
  border: 0;
}

.c9 > button > svg {
  margin: 0 10px;
}

.c1 {
  color: #1B31A8;
  font-size: 20px;
  font-weight: 500;
  padding-top: 48px;
  margin-bottom: 24px;
}

.c5 {
  color: #1E2A32;
  font-size: 24px;
  font-weight: 800;
  margin: 0;
}

.c6 {
  color: #708797;
  font-size: 16px;
  font-weight: 400;
  margin: 10px 0 0 0;
}

.c0 {
  background-color: #F4F8FA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 > div {
  margin-left: 16px;
}

.c8 {
  color: #1E2A32;
  font-size: 14px;
  font-weight: 400;
  margin: 0 0 6px 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: 28px;
}

.c7 > div > div:nth-child(1) > div > div {
  font-family: Rubik,normal;
  font-size: 24px;
  line-height: 120%;
  font-weight: 500;
}

.c7 #input-with-icon-textfield {
  padding: 10px 0;
}

.c14 {
  color: #1E2A32;
  font-size: 20px;
  font-weight: 400;
  margin-left: 32px;
}

.c15 {
  color: #0079FF;
  margin: 24px 30.5px 16px;
  font-size: 32px;
  font-weight: 500;
  text-align: right;
}

.c16 {
  color: #1E2A32;
  margin: 24px 32px;
  font-size: 12px;
  font-weight: 400;
}

.c12 {
  margin-top: 24px;
}

.c13 {
  border: 1px solid #E9EEF2;
  border-radius: 8px;
  background: linear-gradient(180deg,#FFF 50%,#F4F8FA 50%);
}

.c13 > div {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 32px;
}

@media (min-width:768px) {
  .c2 {
    width: 560px;
  }
}

@media (min-width:768px) {
  .c3 {
    padding: 40px;
  }
}

@media (min-width:768px) {
  .c0 {
    height: 90vh;
  }
}

@media (max-width:899px) {
  .c7 > div > div:nth-child(2) {
    margin-top: 16px;
  }
}

<div>
  <section
    class="c0"
  >
    <p
      class="c1"
    >
      Let‘s plan your 
      <b>
        saving goal.
      </b>
    </p>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c2 css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root c3 css-46bh2p-MuiCardContent-root"
      >
        <div
          class="c4"
        >
          <buy-a-house.svg />
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="c5"
            >
              Buy a house
            </p>
            <p
              class="c6"
            >
              Saving goal
            </p>
          </div>
        </div>
        <div
          class="c7"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-md-2 MuiGrid-spacing-lg-2 css-1t4pi4v-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 MuiGrid-grid-lg-7 css-1a4ufwv-MuiGrid-root"
            >
              <p
                class="c8"
              >
                Total amount
              </p>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
                  inputmode="numeric"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                    <dollar-sign.svg />
                  </div>
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                    id="input-with-icon-textfield"
                    inputmode="numeric"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-nnbavb"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 MuiGrid-grid-lg-5 css-urj5pv-MuiGrid-root"
            >
              <p
                class="c8"
              >
                Reach goal by
              </p>
              <div
                class="c9"
                role="monthContent"
                tabindex="0"
              >
                <button
                  role="buttonDecrement"
                >
                  <arrow-left.svg />
                </button>
                <div
                  class="MuiBox-root css-0"
                >
                  <p
                    class="c10"
                    role="paragraphMonth"
                  >
                    December
                  </p>
                  <p
                    class="c11"
                    role="paragraphYear"
                  >
                    2021
                  </p>
                </div>
                <button
                  role="buttonIncrement"
                >
                  <arrow-right.svg />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c12"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 c13 css-ee20y-MuiPaper-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <p
                class="c14"
              >
                Monthly amount
              </p>
              <p
                class="c15"
                role="amountValue"
              >
                <span>
                  $0
                </span>
              </p>
            </div>
            <p
              class="c16"
              role="amountDescription"
            >
              You‘re planning 
              <b>
                1
                 monthly deposits
              </b>
               to reach your
              <b>
                 
                0
              </b>
               goal by 
              <b>
                October 2020
              </b>
              .
            </p>
          </div>
          <div
            class="c17"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root c18 css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="c19"
              >
                Confirm
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
